### ISSUE TITLE
Using an unprivileged user

### ISSUE THREAD

**Issue by Alveel** [bitnami/bitnami-docker-dokuwiki#64]
Considering Docker (and containers in general) best practices, as well as the relative simplicity of running DokuWiki, I think it would be great to make it possible to run this image as a non-root user.

Adding security policies that disallow running containers as root in Kubernetes or OpenShift, and for us it's a requirement :)

Now I'm not too knowledgeable about Bitnami's base images or entrypoints, but it seems simply changing
```patch
diff --git a/0/ol-7/Dockerfile b/0/ol-7/Dockerfile
index 08fdd49..de3182f 100644
--- a/0/ol-7/Dockerfile
+++ b/0/ol-7/Dockerfile
@@ -20,5 +20,7 @@ ENV BITNAMI_APP_NAME="dokuwiki" \
 
 EXPOSE 80 443
 
+USER bitnami
+
 ENTRYPOINT [ "/app-entrypoint.sh" ]
 CMD [ "nami", "start", "--foreground", "apache" ]
```
...is insufficient.

I'd be happy to help where I can, and open to discuss :)

**Comment by alemorcuq** [comment]
Hi, @Alveel.

Unfortunately, we don't currently support non-root containers for Dokuwiki. We have been focused on migrating our infrastructure containers to a non-root approach, such as Apache, MariaDB or Nginx, to name a few. At the moment, we don't know when are we going to start migrating our applications.

Just adding a user to the `Dockerfile` is not enough because, for example, that user doesn't have permissions to work on the files installed by `bitnami-pkg`. You can set the environment variable `BITNAMI_PKG_CHMOD="-R g+rwX"` before using `bitnami-pkg` to give the user permissions on those files, but the logic in those files may need further changes to be correctly executed by the non-root user. All those files are stored under `/root/.nami` by default.

Regards,
Alejandro

**Comment by Alveel** [comment]
Thanks for the info @alemorcuq! Is there a more centralised place where I can follow this progress and potentially contribute?

**Comment by Alveel** [comment]
Aye. After digging in the setup, attempting to figure out how the node application ties in with everything and trying to modify the image to make it work with a non-root user, it's more like hacking a workaround instead of making a proper solution.

**Comment by Alveel** [comment]
Also see my attempt at it anyways: Alveel@fd36c0e808ba9c361f6faaabf10ac06beeb62ec8

**Comment by alemorcuq** [comment]
You can maybe try extending from `bitnami/dokuwiki` to make your `Dockerfile` more simple and also have an easy way to update your image when Bitnami releases a new one.

We understand customizing the container is really hard at the moment. It will get better in the future, so sorry for the inconvenience.

**Comment by Alveel** [comment]
I have started working on a relatively simple docker image that allows non-root (working) and using arbitrary user ID's (WIP) [here](https://github.com/Alveel/docker-dokuwiki-nonroot).

**Comment by marcosbc** [comment]
Hi, we have released version `20180422.4.0-debian-10-r0` which now runs as a non-root user with all the benefits of Bitnami images. Feel free to try it out and let us know what you think!
